tests/cpp/target/parsers/aprofile_test.cc

```diff
@@ -290,9 +290,28 @@ TEST(AProfileParser, ArchVersionInvalidLetter) {
   ASSERT_EQ(Downcast<Bool>(features.at("has_dotprod")), false);
 }
 
+using AProfileOptionalSVE = testing::TestWithParam<float>;
+TEST_P(AProfileOptionalSVE, OptionalSVESupport) {
+  const auto arch_attr = "+v" + std::to_string(GetParam()) + "a";
+
+  // Check that the "has_sve" feature is not set by default when "+sve" isn't set as an attribute.
+  auto target = ParseTargetWithAttrs("", "aarch64-arm-none-eabi", {arch_attr});
+  auto features = Downcast<TargetFeatures>(target.at("features"));
```

Comment:
Can you change these `auto`s to explicit types similar to the other tests?

Reply:
In the first and third case it avoids repeating the type name that is already on the RHS of the assignment operator. The repeated typename is redundant information to the reader and is just extra noise. Anyone reading case 1 can see that it is a `std::string` from the call to `std::to_string` and reading case 2 can see it is a `TargetFeatures` from the call to `Downcast<TargetFeatures>`. I don't have a strong opinion on this, but I do think `auto` adds value here and isn't just avoiding writing an explicit type.

Reply:
Any thoughts on the above, would be good to get this merged since it's been open a while :)

Reply:
I presume the point that @Mousius is making here is that, while auto is fine to use in the way that you are using it, it doesn't match with the way these types have been specified in the rest of the file and, therefore, creates a moment of hesitation when skimming through the code and makes the reader think, "is something different being done here?". If this change was in a new file it would be fine, but being in the middle of a file, where in all other cases these types are specified differently, creates a bit of incongruousness.

Reply:
I've updated this to name the types.